test/e2e/storage/openstack_utils.go

```diff
@@ -74,4 +78,83 @@ func getOpenstackClaimSpecWithStorageClassAnnotation(ns string, diskSize string,
 		},
 	}
 	return claim
-}
\ No newline at end of file
+}
+
+func getOpenstack(c clientset.Interface) (*openstack.OpenStack, error) {
+	os, err := openstack.GetOpenstack()
+	if err != nil {
+		return nil, err
+	}
+	return os, nil
+}
+
+func getOpenstackPodSpecWithVolumeID(volumeID string, keyValuelabel map[string]string, commands []string) *v1.Pod {
+	var volumeMounts []v1.VolumeMount
+	var volumes []v1.Volume
+
+	name := fmt.Sprintf("volume")
+	volumeMounts = append(volumeMounts, v1.VolumeMount{Name: name, MountPath: "/mnt/" + name})
+
+	openstackVolume := new(v1.CinderVolumeSource)
+	openstackVolume.VolumeID = volumeID
+	openstackVolume.FSType = "ext4"
+
+	volumes = append(volumes, v1.Volume{Name: name})
+	volumes[1].VolumeSource.Cinder = openstackVolume
+	
+
+	if commands == nil || len(commands) == 0 {
+		commands = []string{
+			"/bin/sh",
+			"-c",
+			"while true; do sleep 2; done",
+		}
+	}
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "openstack-e2e-",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:         "openstack-e2e-container-" + string(uuid.NewUUID()),
+					Image:        "busybox",
+					Command:      commands,
+					VolumeMounts: volumeMounts,
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+			Volumes:       volumes,
+		},
+	}
+
+	if keyValuelabel != nil {
+		pod.Spec.NodeSelector = keyValuelabel
+	}
+	return pod
+}
+
+func verifyOpenstackDiskAttached(c clientset.Interface, osp *openstack.Openstack, volumeID string) (bool, error) {
+	var (
+		isAttached bool
+		err        error
+	)
+	if osp == nil {
+		osp, err = getOpenstack()
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	instanceId, err := osp.InstanceID()
+
+	isAttached, err = osp.DiskIsAttached(instanceId, volumeID)
+	Expect(err).NotTo(HaveOccurred())
+	return isAttached, err
+}
+
```

Comment:
add comment explaining why `waitOpenstackDiskToDetach` does nothing?

Reply:
Refactored to a new wait function.